mesonbuild/backend/ninjabackend.py

```diff
@@ -1313,78 +1315,70 @@ def generate_swift_target(self, target, outfile):
         else:
             raise MesonException('Swift supports only executable and static library targets.')
 
-    def generate_static_link_rules(self, is_cross, outfile):
+    def generate_static_link_rules(self, outfile):
         num_pools = self.environment.coredata.backend_options['backend_max_links'].value
-        if 'java' in self.build.compilers:
-            if not is_cross:
-                self.generate_java_link(outfile)
-        if is_cross:
-            if self.environment.cross_info.need_cross_compiler():
-                static_linker = self.build.static_cross_linker
-            else:
-                static_linker = self.build.static_linker
-            crstr = '_CROSS'
-        else:
-            static_linker = self.build.static_linker
-            crstr = ''
-        if static_linker is None:
-            return
-        rule = 'rule STATIC%s_LINKER\n' % crstr
-        if static_linker.can_linker_accept_rsp():
-            command_template = ''' command = {executable} $LINK_ARGS {output_args} @$out.rsp
+
+        if 'java' in self.build.for_host.compilers:
+            self.generate_java_link(outfile)
+        ctypes = [
+            (MachineChoice.BUILD,  self.build.for_build.static_linker),
+            (MachineChoice.HOST,   self.build.for_host.static_linker),
+            (MachineChoice.TARGET, self.build.for_target.static_linker)
+        ]
```

Comment:
Try making rules for each platform regardless of what `is_cross` would have been.